backend/session/meson.build

```diff
@@ -21,15 +21,41 @@ if not get_option('logind').disabled()
 	msg += 'You may need to pass "-Dlogind-provider=elogind" or "-Dlogind-provider=systemd" to ensure the correct library is detected.'
 endif
 
-logind = dependency('lib' + get_option('logind-provider'),
-	required: get_option('logind'),
-	not_found_message: '\n'.join(msg).format(get_option('logind-provider')),
-	version: '>=237',
-)
+logind_version = '>=237'
+
+if get_option('logind-provider') == 'auto'
+	assert(get_option('auto_features').auto(), '-Dlogind-provider must be set to systemd or elogind since auto_features != auto')
+	logind = dependency('libsystemd',
+		required: get_option('logind'),
+		not_found_message: 'libsystemd not found, trying libelogind',
```

Comment:
Can you change these `not_found_message`s to match the non-auto case? I wanted them to describe what arguments should be passed and what they're for.

Reply:
Yes, they're added now.